source/common/ssl/connection_impl.cc

```diff
@@ -139,16 +135,25 @@ Network::ConnectionImpl::PostIoAction ConnectionImpl::doHandshake() {
 
 void ConnectionImpl::drainErrorQueue() {
   bool saw_error = false;
+  bool saw_counted_error = false;
```

Comment:
Optional: another option which might be more robust would be for `saw_counted_error_` to be a member of `ConnectionImpl` and, when your callback goes to log it, it goes from `SSL` to `ConnectionImpl` to setting the boolean. But this should work too? *shrug*